extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiConfig.java

```diff
@@ -0,0 +1,409 @@
+    int defaultModelExpandDepth;
+
+    /**
+     * Controls how the model is shown when the API is first rendered.
+     */
+    @ConfigItem(defaultValue = "EXAMPLE")
+    Rendering defaultModelRendering;
+
+    /**
+     * Controls the display of operationId in operations list.
+     */
+    @ConfigItem
+    boolean displayOperationId;
+
+    /**
+     * Controls the display of the request duration (in milliseconds) for Try-It-Out requests.
+     */
+    @ConfigItem
+    boolean displayRequestDuration;
+
+    /**
+     * Controls the default expansion setting for the operations and tags.
+     */
+    @ConfigItem(defaultValue = "LIST")
+    Expansion docExpansion;
+
+    /**
+     * If set, enables filtering. The top bar will show an edit box that could be used to filter
+     * the tagged operations that are shown.
+     */
+    @ConfigItem
+    boolean filter;
+    /**
+     * The name of a component available via the plugin system to use as the top-level layout for Swagger UI.
+     */
+    @ConfigItem(defaultValue = "StandaloneLayout")
+    String layout;
+
+    /**
+     * If set, limits the number of tagged operations displayed to at most this many.
+     */
+    @ConfigItem
+    OptionalInt maxDisplayedTags;
+
+    /**
+     * Controls the display of vendor extension (x-) fields and values.
+     */
+    @ConfigItem
+    boolean showExtensions;
+
+    /**
+     * Controls the display of extensions.
+     */
+    @ConfigItem
+    boolean showCommonExtensions;
+
+    /**
+     * The supported try it out methods.
+     */
+    @ConfigItem
+    Optional<List<HttpMethod>> supportedSubmitMethods;
+
+    /**
+     * Apply a sort to the operation list of each API.
+     */
+    @ConfigItem
+    Optional<Sort> operationsSorter;
+
+    /**
+     * Apply a sort to the tag list of each API.
+     */
+    @ConfigItem
+    Optional<Sort> tagsSorter;
+
+    /**
+     * The url pointing to API definition (the default points to the URI generated by the OpenAPI extension).
+     */
+    @ConfigItem
+    Optional<URI> url;
+
+    /**
+     * The order of the swagger groups which will be displayed when Swagger UI loads.
+     */
+    @ConfigItem(defaultValue = "ASC")
+    Order groupsOrder;
+
+    /**
+     * When used and Topbar plugin is enabled, the url parameter will not be parsed.
+     * Names and URLs must be unique among all items in this array, since they're used as identifiers.
+     * SpecGroupUrl must match the pattern: <url>?group=<name>
+     */
+    @ConfigItem
+    Optional<Set<SpecGroupUrl>> urls;
+
+    /**
+     * The name of the swagger group which will be displayed when Swagger UI loads.
+     */
+    @ConfigItem
+    Optional<String> urlsPrimaryName;
+
+    /**
+     * Set a different validator URL, for example for locally deployed validators.
+     */
+    @ConfigItem
+    Optional<URI> validatorUrl;
+
+    /**
+     * OAuth redirect URL.
+     */
+    @ConfigItem(defaultValue = "/swagger-ui/oauth2-redirect.html")
```

Comment:
I think you cannot define the defaultValue here. You need to make it optional and create it from the path if not defined as the path is customizable.

Also I would make it a String not a URI.

Reply:
@gsmet , would ou transfom all URI to String ?